fastdns/fastdns.go

```diff
@@ -0,0 +1,33 @@
+// Package fastdsn adapts the lego FastDNS DNS provider
+// for Caddy. Importing this package plugs it in.
+package fastdns
+
+import (
+	"errors"
+
+	"github.com/mholt/caddy/caddytls"
+	"github.com/xenolf/lego/providers/dns/fastdns"
+)
+
+func init() {
+	caddytls.RegisterDNSProvider("fastdns", NewDNSProvider)
+}
+
+// NewDNSProvider returns a new FastDNS DNS challenge provider.
+// The credentials are interpreted as follows:
+//
+// len(0): use credentials from environment
+// len(4): credentials[0] = Akamai Host
+//         credentials[1] = Client Token
+//         credentials[3] = Client Secret
+//         credentials[4] = Access Token
+func NewDNSProvider(credentials ...string) (caddytls.ChallengeProvider, error) {
+	switch len(credentials) {
+	case 0:
+		return fastdns.NewDNSProvider()
+	case 2:
```

Comment:
Oops, this should be `case 4` :)

Reply:
Ooops, brainfart :D It is fixed now I think.